### PR TITLE
Fix wrong messages during MWP Flow

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-1857
Fixes APP-1854

## What changed (plus any additional context for devs)
A few things were happening with MWP causing this issue:

- We were incorrectly handling the keychain errors so any cancellation would trigger the alert about phones swapping.

- Sometimes (?) we were looking up the pkey using a lowercase version while the original is stored checksummed resulting in "unavailable" error.  I am not sure why but I believe this was on Warpcast side sending different versions of the address, which is why reconnecting fixed the issue (they sent the right one there) but somewhere else they were breaking it.  We're now checksumming it so we don't care anymore.

- We were passing through a `chainId` parameter that was making the L1 fee estimation fail

## Screen recordings / screenshots

https://github.com/user-attachments/assets/eab2408d-4304-4268-8850-4b9dd37fd1e6

## What to test

- MWP normal tx signing flow
- MPW cancelling authentication during face id and later pin.
- Normal sends with and without faceID cancellation

### CODE REVIEW

**Review by jinchung — APPROVED** [review]
🌮

**Review by walmat — APPROVED** [review]
legend. tysm @brunobar79

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/add-more-logging-pkey/f46b6e9e41f06ba7d17028b0b9884e6ee52f1b25.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/add-more-logging-pkey/f46b6e9e41f06ba7d17028b0b9884e6ee52f1b25.ipa) for f46b6e9e41f06ba7d17028b0b9884e6ee52f1b25

File: src/keychain/index.ts [hunk 1]
```diff
@@ -170,9 +171,16 @@ export async function get(key: string, options: KeychainOptions = {}): Promise<R
...
-            });
+            // Avoid logging user cancelled operations
+            if (!(e.toString().includes('code: 10') || e.toString().includes('code: 11'))) {
+              logger.error(new RainbowError(`[keychain]: _get() handled unknown error`), {
+                message: e.toString(),
+              });
+              return {
+                value: undefined,
+                error: ErrorType.NotAuthenticated,
+              };
```
Comment (brunobar79) [inline src/keychain/index.ts:182]:
```suggestion
```

File: src/keychain/index.ts [hunk 2]
```diff
@@ -170,9 +171,12 @@ export async function get(key: string, options: KeychainOptions = {}): Promise<R
             return _get(attempts + 1);
           }
           default: {
-            logger.error(new RainbowError(`[keychain]: _get() handled unknown error`), {
-              message: e.toString(),
-            });
+            // Avoid logging user cancelled operations
+            if (!(e.toString().includes('code: 10') || e.toString().includes('code: 11'))) {
```
Comment (jinchung) [inline src/keychain/index.ts:175]:
```suggestion
            if (!(e.toString().includes('code: 10') || e.toString().includes('code: 13'))) {
```

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/add-more-logging-pkey/5b59cdffaff5cf8f7f8515002e674acddb4d684f.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/add-more-logging-pkey/5b59cdffaff5cf8f7f8515002e674acddb4d684f.ipa) for 5b59cdffaff5cf8f7f8515002e674acddb4d684f

**Pull request merged**